.github/ISSUE_TEMPLATE/enhancement.yml

```diff
@@ -0,0 +1,61 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Enhancement
+description: Add new feature, improve code, and more
+labels: [ "type/enhancement" ]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you very much for your enhancement for Pulsar!
+  - type: checkboxes
+    attributes:
+      label: Search before asking
+      description: >
+        Please make sure to search in the [issues](https://github.com/apache/pulsar/issues) first to see whether the same issue was reported already.
+      options:
+        - label: >
+            I had searched in the [issues](https://github.com/apache/pulsar/issues) and found no similar issues.
+          required: true
+  - type: textarea
+    attributes:
+      label: Motivation
+      description: Describe the motivation you'd like to make this enhancement.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Solution
+      description: Describe the proposed solution and add related materials like links if any.
+  - type: textarea
+    attributes:
+      label: Alternatives
+      description: Describe the any alternative solutions or features you've considered.
```

Comment:
`Describe other alternative solutions or features you considered, but rejected.`